lib/view_model/buy/buy_sell_view_model.dart

```diff
@@ -409,44 +446,67 @@ abstract class BuySellViewModelBase extends WalletChangeListenerViewModel with S
           Duration(seconds: 10),
           onTimeout: () => null,
         )));
+    final resultSepa = await Future.wait<List<Quote>?>(validProvidersSepa.map((element) => element
+        .fetchQuote(
+          cryptoCurrency: cryptoCurrency,
+          fiatCurrency: FiatCurrency.eur,
+          amount: amount,
+          paymentType: selectedPaymentMethod?.paymentMethodType,
+          isBuyAction: isBuyAction,
+          walletAddress: wallet.walletAddresses.address,
+        )
+        .timeout(
+          Duration(seconds: 10),
+          onTimeout: () => null,
+        )));
 
     sortedRecommendedQuotes.clear();
     sortedQuotes.clear();
 
-    final validQuotes = result
+    final validQuotesNative = result
         .where((element) => element != null && element.isNotEmpty)
         .expand((element) => element!)
         .toList();
 
-    if (validQuotes.isEmpty) {
+    final validQuotesSepa = resultSepa
+        .where((element) => element != null && element.isNotEmpty)
+        .expand((element) => element!)
+        .toList();
+
+    if (validQuotesNative.isEmpty && validQuotesSepa.isEmpty) {
       buySellQuotState = BuySellQuotFailed();
       return;
     }
 
-    validQuotes.sort((a, b) => a.rate.compareTo(b.rate));
+    validQuotesNative.sort((a, b) => a.rate.compareTo(b.rate));
+    validQuotesSepa.sort((a, b) => a.rate.compareTo(b.rate));
 
     final Set<String> addedProviders = {};
-    final List<Quote> uniqueProviderQuotes = validQuotes.where((element) {
+
+    final List<Quote> uniqueProviderQuotesNative = validQuotesNative.where((element) {
```

Comment:
and I feel like there is something missing in this part and the one below, but I am not sure, so I will let @Serhii-Borodenko chime in

Reply:
So far, the only issue I see is with the “Recommended” section. With the current implementation, only successRateQuotes are added there, which means only Onramper quotes with the successRate label. And if we don’t want to duplicate providers that already have native quotes with SEPA quotes in the ‘Recommended’ section, then we don’t need uniqueProviderQuotesSepa

Reply:
Yeah it is messed up, I'll resolve this